crates/primitives/transactions/src/to_starknet_core_transaction.rs

```diff
@@ -133,27 +254,95 @@ pub fn to_starknet_core_tx(
+                        .iter()
+                        .map(|x| Felt252Wrapper::from(*x).into())
+                        .collect::<Vec<FieldElement>>(),
+                    nonce: Felt252Wrapper::from(nonce.0).into(),
+                    sender_address: Felt252Wrapper::from(sender_address.0).into(),
+                    calldata: calldata.0.iter().map(|x| Felt252Wrapper::from(*x).into()).collect::<Vec<FieldElement>>(),
+                }),
+                InvokeTransaction::V3(InvokeTransactionV3 {
+                    resource_bounds,
+                    tip,
+                    signature,
+                    nonce,
+                    sender_address,
+                    calldata,
+                    nonce_data_availability_mode,
+                    fee_data_availability_mode,
+                    paymaster_data,
+                    account_deployment_data,
+                }) => starknet_core::types::InvokeTransaction::V3(starknet_core::types::InvokeTransactionV3 {
+                    transaction_hash,
+                    resource_bounds: api_resources_to_core_ressources(resource_bounds),
+                    tip: tip.0,
+                    signature: signature
+                        .0
+                        .iter()
+                        .map(|x| Felt252Wrapper::from(*x).into())
+                        .collect::<Vec<FieldElement>>(),
+                    nonce: Felt252Wrapper::from(nonce.0).into(),
+                    sender_address: Felt252Wrapper::from(sender_address.0).into(),
+                    calldata: calldata.0.iter().map(|x| Felt252Wrapper::from(*x).into()).collect::<Vec<FieldElement>>(),
+                    nonce_data_availability_mode: api_da_to_core_da(nonce_data_availability_mode).unwrap(),
+                    fee_data_availability_mode: api_da_to_core_da(fee_data_availability_mode).unwrap(),
+                    paymaster_data: paymaster_data
+                        .0
+                        .iter()
+                        .map(|x| Felt252Wrapper::from(*x).into())
+                        .collect::<Vec<FieldElement>>(),
+                    account_deployment_data: account_deployment_data
+                        .0
+                        .iter()
+                        .map(|x| Felt252Wrapper::from(*x).into())
+                        .collect::<Vec<FieldElement>>(),
                 }),
             };
 
             starknet_core::types::Transaction::Invoke(tx)
         }
-        super::Transaction::L1Handler(tx) => {
+        Transaction::L1Handler(tx) => {
             let tx = starknet_core::types::L1HandlerTransaction {
                 transaction_hash,
-                version: 0,
-                nonce: tx.nonce,
-                contract_address: tx.contract_address.into(),
-                entry_point_selector: tx.entry_point_selector.into(),
-                calldata: cast_vec_of_felt_252_wrappers(tx.calldata),
+                version: FieldElement::ZERO,
```

Comment:
Why do we use  `FieldElement::ZERO` as default version, is this the only valid value possible?

Reply:
Actually its seems there is no other version for DeployTx and L1HandlerTx